From cc348256d337cdf5be85acf98ca93fe66c4d0352 Mon Sep 17 00:00:00 2001
From: ninja- <xninja@openmailbox.org>
Date: Tue, 8 Oct 2013 14:35:58 +0200
Subject: [PATCH] Add respawn API.


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 4d555ba8..457aa80c 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1674,6 +1674,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
         public void setCollidesWithEntities(boolean collides) {
             throw new UnsupportedOperationException("Not supported yet.");
         }
+
+        /**
+         * Respawns the player if dead.
+         */
+        public void respawn() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
     }
 
     @NotNull
-- 
2.34.1

