From 3512678f7e2fbcb98620fb10142ad4744c1465a0 Mon Sep 17 00:00:00 2001
From: DerFlash <bte@freenet.de>
Date: Tue, 9 Jul 2013 00:11:12 +0200
Subject: [PATCH] Save ticks lived to nbttag


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 21ce754a1..180381fb2 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -1778,6 +1778,7 @@ public abstract class Entity implements INamableTileEntity, EntityAccess, IComma
             if (maxAirTicks != getDefaultMaxAirSupply()) {
                 nbttagcompound.putInt("Bukkit.MaxAirSupply", getMaxAirSupply());
             }
+            nbttagcompound.putInt("Spigot.ticksLived", this.tickCount);
             // CraftBukkit end
             IChatBaseComponent ichatbasecomponent = this.getCustomName();
 
@@ -1934,6 +1935,11 @@ public abstract class Entity implements INamableTileEntity, EntityAccess, IComma
             }
 
             // CraftBukkit start
+            // Spigot start
+            if (this instanceof EntityLiving) {
+                this.tickCount = nbttagcompound.getInt("Spigot.ticksLived");
+            }
+            // Spigot end
             this.persist = !nbttagcompound.contains("Bukkit.persist") || nbttagcompound.getBoolean("Bukkit.persist");
             // SPIGOT-6907: re-implement LivingEntity#setMaximumAir()
             if (nbttagcompound.contains("Bukkit.MaxAirSupply")) {
-- 
2.34.1

