From d3bd4a3900b1568721b2f6ea97dd85c50ca441f6 Mon Sep 17 00:00:00 2001
From: Thinkofdeath <thethinkofdeath@gmail.com>
Date: Wed, 5 Mar 2014 20:27:27 +0000
Subject: [PATCH] Prevent hoppers from loading chunks


diff --git a/src/main/java/net/minecraft/world/level/block/entity/TileEntityHopper.java b/src/main/java/net/minecraft/world/level/block/entity/TileEntityHopper.java
index d35ce34b2..3111fecb1 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/TileEntityHopper.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/TileEntityHopper.java
@@ -464,6 +464,7 @@ public class TileEntityHopper extends TileEntityLootable implements IHopper {
     private static IInventory getContainerAt(World world, double d0, double d1, double d2) {
         Object object = null;
         BlockPosition blockposition = new BlockPosition(d0, d1, d2);
+        if ( !world.hasChunkAt( blockposition ) ) return null; // Spigot
         IBlockData iblockdata = world.getBlockState(blockposition);
         Block block = iblockdata.getBlock();
 
-- 
2.25.1

